lib/auth/github.go

```diff
@@ -491,9 +491,9 @@ func validateGithubAuthCallbackHelper(ctx context.Context, m githubManager, diag
 }
 
 func (a *Server) getGithubConnector(ctx context.Context, request types.GithubAuthRequest) (types.GithubConnector, error) {
-	if request.SSOTestFlow {
+	if request.SSOTestFlow || request.AuthenticatedUser != "" {
```

Comment:
Is this change needed? In `validateGithubAuthCallback` below we diverge the flow when `AuthenticatedUser` is set, calling `validateGithubAuthCallbackForAuthenticatedUser` instead, so `getGithubConnector` shouldn't be called in the end?

Reply:
this function is also called by `CreateGithubAuthRequest`